src/lib/snark_worker/functor.ml

```diff
@@ -234,6 +234,10 @@ module Make (Inputs : Intf.Inputs_intf) :
     let%bind state =
       Worker_state.create ~constraint_constants ~proof_level ()
     in
+    let constraint_digests =
+      Transaction_snark.constraint_system_digests ~constraint_constants ()
+      |> List.map ~f:(Tuple2.map_snd ~f:Md5_lib.to_binary)
```

Comment:
I wonder, why do you only do this operation on the second element of the tuple? Aren't both elements in the tuple the same type?

Reply:
First element of the tuple is `string` representing the key name. Second element of the tuple is `Md5_lib.t`, which doesn't have `bin_io` definitions. Instead of coming up with a wrapper type, I decided it to be easier to just convert the second element to `string`.